xbmc/messaging/ApplicationMessenger.cpp

```diff
@@ -101,7 +118,7 @@ int CApplicationMessenger::SendMsg(ThreadMessage&& message, bool wait)
     message.result = std::make_shared<int>(-1);
     // check that we're not being called from our application thread, else we'll be waiting
     // forever!
-    if (!g_application.IsCurrentThread())
+    if (!CThread::IsCurrentThread(m_threadID))
```

Comment:
wouldn't it be better to call `m_threadID` something like `m_guiThreadID` so that the code is reading more obvious?